app/javascript/react/components/ThresholdConfigurator/ThresholdConfigurator.tsx

```diff
@@ -0,0 +1,153 @@
+import React, { useEffect, useRef, useState } from "react";
+
+import * as S from "./ThresholdConfigurator.style";
+
+const calculateThumbPercentage = (
+  value: number,
+  min: number,
+  max: number
+): number => {
+  const percentage = (value - min) / (max - min);
+  return percentage;
+};
+
+const calculateThumbPosition = (
+  value: number,
+  min: number,
+  max: number,
+  width: number
+): number => {
+  const percentage = calculateThumbPercentage(value, min, max);
+  return percentage * width;
+};
+
+interface ThresholdsConfiguratorProps {
+  initialThresholds: number[];
+}
+
+const ThresholdsConfigurator: React.FC<ThresholdsConfiguratorProps> = ({
+  initialThresholds,
+}) => {
+  const [thresholds, setThresholds] = useState(initialThresholds);
+  const [sliderWidth, setSliderWidth] = useState(0);
+  const [thumbPositions, setThumbPositions] = useState<number[]>([]);
```

Comment:
Same as for `thresholds`, I believe we can make it and use using interface and making less error-prone